arm/Microsoft.Network/applicationGateways/deploy.bicep

```diff
@@ -151,244 +240,108 @@ var diagnosticsMetrics = [for metric in metricsToEnable: {
-          }
-        }
-      }
-    ]
-    sslCertificates: empty(sslCertificateKeyVaultSecretId) ? null : sslCertificates
+  identity: identity
+  properties: union({
+    authenticationCertificates: authenticationCertificates
+    autoscaleConfiguration: autoscaleMaxCapacity > 0 && autoscaleMinCapacity > 0 ? {
+      maxCapacity: autoscaleMaxCapacity
+      minCapacity: autoscaleMinCapacity
+    } : null
     backendAddressPools: backendAddressPools
-    probes: probes_var
-    backendHttpSettingsCollection: backendHttpConfigurations_var
+    backendHttpSettingsCollection: backendHttpSettingsCollection
+    customErrorConfigurations: customErrorConfigurations
+    enableHttp2: enableHttp2
+    firewallPolicy: !empty(firewallPolicyId) ? {
+      id: firewallPolicyId
+    } : null
+    forceFirewallPolicyAssociation: !empty(firewallPolicyId)
+    frontendIPConfigurations: frontendIPConfigurations
     frontendPorts: frontendPorts
+    gatewayIPConfigurations: gatewayIPConfigurations
+    globalConfiguration: {
+      enableRequestBuffering: enableRequestBuffering
+      enableResponseBuffering: enableResponseBuffering
+    }
     httpListeners: httpListeners
+    loadDistributionPolicies: loadDistributionPolicies
+    privateLinkConfigurations: privateLinkConfigurations
+    probes: probes
     redirectConfigurations: redirectConfigurations
     requestRoutingRules: requestRoutingRules
-    enableHttp2: http2Enabled
-    webApplicationFirewallConfiguration: startsWith(sku, 'WAF') ? wafConfiguration : null
-  }
-  dependsOn: []
+    rewriteRuleSets: rewriteRuleSets
+    sku: {
+      name: sku
+      tier: endsWith(sku, 'v2') ? sku : substring(sku, 0, indexOf(sku, '_'))
+      capacity: autoscaleMaxCapacity > 0 && autoscaleMinCapacity > 0 ? null : capacity
+    }
+    sslCertificates: sslCertificates
+    sslPolicy: {
+      cipherSuites: sslPolicyCipherSuites
+      // disabledSslProtocols: sslPolicyDisabledSslProtocols
+      minProtocolVersion: sslPolicyMinProtocolVersion
+      policyName: empty(sslPolicyName) ? null : sslPolicyName
+      policyType: sslPolicyType
+    }
+    sslProfiles: sslProfiles
+    trustedClientCertificates: trustedClientCertificates
+    trustedRootCertificates: trustedRootCertificates
+    urlPathMaps: urlPathMaps
+    webApplicationFirewallConfiguration: webApplicationFirewallConfiguration
+    // {
+    //   disabledRuleGroups: [
+    //     {
+    //       ruleGroupName: 'string'
+    //       rules: [
+    //         int
+    //       ]
+    //     }
+    //   ]
+    //   enabled: bool
+    //   exclusions: [
+    //     {
+    //       matchVariable: 'string'
+    //       selector: 'string'
+    //       selectorMatchOperator: 'string'
+    //     }
+    //   ]
+    //   fileUploadLimitInMb: int
+    //   firewallMode: 'string'
+    //   maxRequestBodySize: int
+    //   maxRequestBodySizeInKb: int
+    //   requestBodyCheck: bool
+    //   ruleSetType: 'string'
+    //   ruleSetVersion: 'string'
+    // }
```

Comment:
Is this to be removed?

Reply:
Ooopsie. Have to look into that one. Sorry

Reply:
fixed it